internal/service/ipam/api_server_resource_test.go

```diff
@@ -0,0 +1,1089 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServerExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ddns_client_update", "client"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccServerDdnsClientUpdate(name, "server"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServerExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ddns_client_update", "server"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+// issue
+func TestAccServerResource_DdnsConflictResolutionMode(t *testing.T) {
+	var resourceName = "bloxone_dhcp_server.test_ddns_conflict_resolution_mode"
+	var v ipam.IpamsvcServer
+	var name = acctest.RandomNameWithPrefix("dhcp-server")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccServerDdnsConflictResolutionMode(name, false, "check_exists_with_dhcid"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServerExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ddns_use_conflict_resolution", "false"),
+					resource.TestCheckResourceAttr(resourceName, "ddns_conflict_resolution_mode", "check_exists_with_dhcid"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccServerDdnsConflictResolutionMode(name, true, "check_with_dhcid"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServerExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ddns_use_conflict_resolution", "true"),
+					resource.TestCheckResourceAttr(resourceName, "ddns_conflict_resolution_mode", "check_with_dhcid"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccServerResource_DdnsDomain(t *testing.T) {
+	var resourceName = "bloxone_dhcp_server.test_ddns_domain"
+	var v ipam.IpamsvcServer
+	var name = acctest.RandomNameWithPrefix("dhcp-server")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccServerDdnsDomain(name, "Test.com."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServerExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ddns_domain", "Test.com."),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccServerDdnsDomain(name, "Test_update.com."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServerExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ddns_domain", "Test_update.com."),
```

Comment:
`_` should not be used in a domain name.
```suggestion
					resource.TestCheckResourceAttr(resourceName, "ddns_domain", "test-update.com."),
```

Reply:
done